Private nested record cannot be serialized

**Describe the bug**
I am getting this exception:

```
com.esotericsoftware.kryo.KryoException: java.lang.IllegalAccessException: class com.esotericsoftware.kryo.serializers.RecordSerializer cannot access a member of class mypkg.A$B with modifiers "public"
	at com.esotericsoftware.kryo.serializers.RecordSerializer.componentValue(RecordSerializer.java:208)
	at com.esotericsoftware.kryo.serializers.RecordSerializer.write(RecordSerializer.java:92)
	at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:642)
	at com.esotericsoftware.kryo.serializers.ReflectField.write(ReflectField.java:70)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:108)
	at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:642)
	at com.esotericsoftware.kryo.serializers.ReflectField.write(ReflectField.java:70)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:108)
	at com.esotericsoftware.kryo.Kryo.writeClassAndObject(Kryo.java:711)
	...
Caused by: java.lang.IllegalAccessException: class com.esotericsoftware.kryo.serializers.RecordSerializer cannot access a member of class mypkg.A$B with modifiers "public"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:392)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674)
	at java.base/java.lang.reflect.Method.invoke(Method.java:560)
	at com.esotericsoftware.kryo.serializers.RecordSerializer.componentValue(RecordSerializer.java:206)
	... 19 more
```

**To Reproduce**
```java
public class A {
    private static record B(String y) {}
    private B x;
}

// serialized with:
kryo.writeClassAndObject(output, someA);
```

**Environment:**
 - OS: Linux
 - JDK Version: 17
 - Kryo Version: 5.2.1

**Additional context**
There's no exception when the record is public.

This test fails:

```java
public class KryoRecordTest {
	static class A {
		private record B(String y) {
		}
		private B x;
		public A() {
			x = new B("w");
		}
	}
	@Test
	void test() throws IOException {
		try (	var stream = new ByteArrayOutputStream();
			var output = new Output(stream)) {
			ThreadLocalKryo.get().writeClassAndObject(output, new A());
		}
	}
}
```

Exception:

```
com.esotericsoftware.kryo.KryoException: 
java.lang.IllegalAccessException: class com.esotericsoftware.kryo.serializers.RecordSerializer cannot access a member of class mypkg.KryoRecordTest$A$B with modifiers "public"
Serialization trace:
Could not retrieve record components (mypkg.KryoRecordTest$A$B)
y (java.lang.String)
x (mypkg.KryoRecordTest$A)
	at mypkg.KryoRecordTest.test(KryoRecordTest.java:21)
Caused by: java.lang.IllegalAccessException: class com.esotericsoftware.kryo.serializers.RecordSerializer cannot access a member of class mypkg.KryoRecordTest$A$B with modifiers "public"
	at mypkg.KryoRecordTest.test(KryoRecordTest.java:21)
```

From pom.xml:

```xml
	<properties>
		<maven.compiler.release>17</maven.compiler.release>
		...
	</properties>

	<dependencies>
		<dependency>
			<groupId>com.esotericsoftware</groupId>
			<artifactId>kryo</artifactId>
			<version>5.2.1</version>
		</dependency>
		...
	</dependencies>

	<build>
		<plugins>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>3.8.1</version>
				<configuration>
					<compilerArgs>
						<compilerArg>-parameters</compilerArg>
						<compilerArg>--enable-preview</compilerArg>
					</compilerArgs>
				</configuration>
			</plugin>
			<plugin>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>3.0.0-M5</version>
				<configuration>
					<argLine>--enable-preview</argLine>
				</configuration>
			</plugin>
		</plugins>
	</build>
```

There is currently no `moduie-info.java` in the project, i.e. it's an unnamed module.

--- a/src/com/esotericsoftware/kryo/serializers/RecordSerializer.java
+++ b/src/com/esotericsoftware/kryo/serializers/RecordSerializer.java
@@ -203,6 +203,9 @@ public class RecordSerializer<T> extends ImmutableSerializer<T> {
 		RecordComponent recordComponent) {
 		try {
 			Method get = recordObject.getClass().getDeclaredMethod(recordComponent.name());
+			if (!get.canAccess(recordObject)) {
+				get.setAccessible(true);
+			}
 			return get.invoke(recordObject);
 		} catch (Throwable t) {
 			KryoException ex = new KryoException(t);

